File: Vinayak-PB002/phone_number_validator/phone.rb
phone_number = ""
if ARGV.length == 0
	puts "Enter phone number :"
	phone_number = gets.chomp
	puts phone_number
else
	phone_number = ARGV[0]
	puts phone_number
end
VALID_PHONE_NUMBER_REGEX = /\A([0]|(\+91))?[789]\d{9}\Z/
if VALID_PHONE_NUMBER_REGEX =~ phone_number
	puts "Valid phone Number!"
else
	puts "Invalid phone number!"
end

File: Rishu-PB002/phone/phone.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;
import java.io.*;
import java.util.Scanner;


public class phone{

    public static void main(String[] argv) {
        String phoneNumber;

        if (argv.length==0){
             Scanner sc = new Scanner(System.in);
             System.out.println("Please enter a Number :");
             phoneNumber=sc.nextLine();
        }
        else{
            phoneNumber = argv[0];
        }   
        System.out.println(phoneNumber);
        // regex for Indian numbers
        String regex = "^(((\\+|00)\\d{1,3}[\\s-]?)|0)?\\d{10}$";
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(phoneNumber);

        if (matcher.matches()) {
            System.out.println("Phone Number Valid");
        } else {
            System.out.println("Phone Number must be in the indian format");
        }
    }
}